microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+
+
+<!-- ====================================================================== -->
+## Load content scripts only when needed
+
+To reduce the amount of code that is loaded and that runs on each webpage, and to save memory and CPU resources, only load your content scripts as needed, instead of on every page load.
+
+
+<!-- ------------------------------ -->
+#### Configure when to load content scripts in your extension manifest file
+
+To control when your extension's content script should be loaded, use the `run_at` property in your extension's manifest file.
+
+By default, this property is set to the `document_idle` value, which means that the content script will be loaded, and run, after the page has finished loading and the DOM is ready. This is the recommended value for most content scripts. The `document_idle` value ensures that the content script will not interfere with the page loading process.
+
+To load and run your content script before the page is fully loaded, use the `document_start` or `document_end` values. These values are useful in cases such as modifying the webpage layout or styling, but they can also cause performance issues or compatibility problems with other scripts on the page.
+
+
+<!-- ------------------------------ -->
+#### Programmatically load content scripts at runtime
+
+To programmatically load your content scripts at runtime, only when needed, use the `chrome.scripting` API . The `chrome.scripting` API provides more control over when and where your content script is loaded.
+
+For example, you can use the `chrome.scripting` API to load your content script only after the user has interacted with the webpage or extension UI, such as when clicking on an extension button, or a part of the webpage.
+
+If you use the `chrome.scripting` API when the user interacts with the webpage, make sure to carefully consider if you need to repeatedly load your content script every time the interaction happens. Loading content scripts too frequently can cause user experience issues or errors.
+
+
+<!-- ====================================================================== -->
+## Avoid blocking calls or long running synchronous tasks
+
+Blocking calls and long-running synchronous tasks can delay the webpage load or make other aspects of a webpage slower, as well as negatively impact the UI responsiveness.
+
+Blocking calls are JavaScript operations that prevent running other code until they complete. For example, using the `XMLHttpRequest`, `localStorage`, or `chrome.storage.sync` APIs, which are synchronous, prevents the webpage from running other code.
+
+Long running synchronous tasks are those that take a long time to complete, preventing the browser from running other webpage code while they are running. This can include complex calculations, loops, or string manipulations.
```

Comment:
```suggestion
_Long-running synchronous tasks_ are those that take a long time to complete, preventing the browser from running other webpage code while they are running. This can include complex calculations, loops, or string manipulations.
```